src/api/libcellml/variable.h

```diff
@@ -242,30 +254,21 @@ class LIBCELLML_EXPORT Variable: public NamedEntity
      *
      * Tests to see if the argument variable exists in the set of this variable's equivalent
      * variables. Returns @c true if the argument variable is in this variable's equivalent
-     * variables and @c false otherwise.
-     *
-     * @param equivalentVariable The variable to check for in this variable's equivalent variables.
+     * variables and @c false otherwise.  By default the test will *not* traverse the equivalent
+     * network to determine if the two variables are equivalent.
      *
-     * @return @c true if the @p equivalentVariable is in this variable's equivalent variables
-     * and @c false otherwise.
+     * If the optional parameter @p considerIndirectEquivalences is @c true then the test *will*
+     * consider the entire equivalence network that this variable is a part of.
      *
-     * @deprecated This method is currently needed by our Printer and Validator classes, but this is
-     * not a method that a libCellML user should ever need. It will therefore be removed at some point.
-     */
-    bool hasDirectEquivalentVariable(const VariablePtr &equivalentVariable) const;
-
-    /**
-     * @brief Test whether the argument variable is equivalent to this variable.
-     *
-     * Tests to see if the argument variable is equivalent to this variable. Returns @c true if
-     * the argument variable is equivalent to this variable and @c false otherwise.
-     *
-     * @param equivalentVariable The variable to check for equivalence.
+     * @param equivalentVariable The variable to check for in this variable's equivalent variables.
+     * @param considerIndirectEquivalences Optional parameter to expand the test to the entire
+     * equivalent network of this variable.
      *
-     * @return @c true if the @p equivalentVariable is equivalent to this variable and
-     * @c false otherwise.
+     * @return @c true if the @p equivalentVariable is in this variable's equivalent variables set or
+     * if @p considerIndirectEquivalences is @c true then @c true if the @p equivalentVariable is in
+     * this variable's equivalence network.  In all other cases @c false is returned.
      */
-    bool hasEquivalentVariable(const VariablePtr &equivalentVariable) const;
+    bool hasEquivalentVariable(const VariablePtr &equivalentVariable, bool considerIndirectEquivalences = false) const;
 
```

Comment:
I thought it was the other way around (so checks for indirect connections by default)?

Reply:
No that changed as per your comments above.  The default position is to now consider only directly equivalent variables and then optionally indirect variables.

Reply:
OK